code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -50,6 +54,10 @@
 /obj/item/weapon/reagent_containers/food/drinks/afterattack(obj/target, mob/user , proximity)
 	if(!proximity) return
 	if(istype(target, /obj/structure/reagent_dispensers)) //A dispenser. Transfer FROM it TO us.
+	
+		if (!is_open_container())
+			to_chat(user, "<span class='warning'>[target]'s tab isn't open!</span>")
```

Comment:
Ok, even though there should be no situation where that triggers for non-soda, barring badminnery/varediting